test/clamm.test.ts

```diff
@@ -212,19 +212,143 @@ describe('math tests', () => {
+
+      expect(resultUp.returns).toEqual(1446690239n)
+      expect(resultDown.returns).toEqual(1446690238n)
+    })
+
+    test('big', async () => {
+      const sqrtPriceA = 1_000000000000000000000000n
+      const sqrtPriceB = 2_000000000000000000000000n
+      const liquidity = (2n ** 64n - 1n) * 1_00000n
+
+      const paramsUp = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: true } }
+      const paramsDown = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: false } }
+      const resultUp = await clamm.methods.getDeltaY(paramsUp)
+      const resultDown = await clamm.methods.getDeltaY(paramsDown)
+
+      expect(resultUp.returns).toEqual(liquidity / 1_00000n)
+      expect(resultDown.returns).toEqual(liquidity / 1_00000n)
+    })
+
+    test('overflow', async () => {
+      const sqrtPriceA = 1_000000000000000000000000n
+      const sqrtPriceB = 2n ** 256n - 1n
+      const liquidity = 2n ** 256n - 1n
+
+      const paramsUp = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: true } }
+      const paramsDown = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: false } }
+      await expectError(clamm.methods.getDeltaY(paramsUp))
+      await expectError(clamm.methods.getDeltaY(paramsDown))
+    })
+
+    test('huge liquidity', async () => {
+      const sqrtPriceA = 1_000000000000000000000000n
+      const sqrtPriceB = 1_000000000000000001000000n
+      const liquidity = 2n ** 256n - 1n
+
+      const paramsUp = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: true } }
+      const paramsDown = { args: { sqrtPriceA, sqrtPriceB, liquidity, roundingUp: false } }
+      const resultUp = await clamm.methods.getDeltaY(paramsUp)
+      const resultDown = await clamm.methods.getDeltaY(paramsDown)
+
+      expect(resultUp.returns).toEqual(1157920892373161954235709850086879078532699846656405641n)
+      expect(resultDown.returns).toEqual(1157920892373161954235709850086879078532699846656405640n)
+    })
   })
+
+  describe('get delta y - domain', () => {
+    let clamm: CLAMMInstance
+    const minSqrtPrice = 15258932000000000000n
+    const maxSqrtPrice = 65535_383934512647000000000000n
+    const minLiquidity = 1n
+    const maxLiquidity = 2n ** 256n - 1n
+
+    beforeEach(async () => {
+      clamm = (await deployCLAMM(sender)).contractInstance
+    })
+
+    it('maximize delta sqrt price and liquidity', async () => {
+      const paramsUp = {
+        args: { sqrtPriceA: maxSqrtPrice, sqrtPriceB: minSqrtPrice, liquidity: maxLiquidity, roundingUp: true }
+      }
+      const paramsDown = {
+        args: { sqrtPriceA: maxSqrtPrice, sqrtPriceB: minSqrtPrice, liquidity: maxLiquidity, roundingUp: false }
+      }
+      const resultUp = await clamm.methods.getDeltaY(paramsUp)
+      const resultDown = await clamm.methods.getDeltaY(paramsDown)
+
+      expect(resultUp.returns).toEqual(75884790229800029582010010030152469040784228171629896039591333116952600000001n)
```

Comment:
It seems there are incorrect results

Reply:
have to wait for div refactor pr to be merged